serverless-heaven/serverless-webpack
# Convert packageModules to use bestzip instead of archiver

## What did you implement:
Updated packageModules to use bestzip, allowing the module zip to use the native zip function on the server instead of archiver, if it exists. This greatly speeds up the build time when zipping archives. In our case, it changed from about 25 seconds per function to about 1.5 seconds per function.

Closes #561 



## How did you implement it:
Replaced the code that uses archiver for bestzip, and tested it to make sure that the resultant zip created by bestzip has the same file structure as the zips currently created by archiver. Bestzip falls back to using archiver, so if zip is not installed on the server it should still function as it does today.

To get the unit tests to pass, I created a bestzip mock, similar to what was previously done for archiver. I think it makes sense as a future enhancement to actually test that the zip is working as expected instead of just mocking it.



## How can we verify it:
To verify it needs to be tested by running a package build on a server that contains a native zip function as well as on a server that doesn't, to make sure that both paths of bestzip work.


## Todos:


***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO
